File: python/twelve-days/twelve_days.py
def recite(start_verse, end_verse):
    """
    Generate the lyrics of 'The Twelve Days of Christmas' song.

    Args:
        start_verse (int): The starting verse (1-12).
        end_verse (int): The ending verse (1-12).

    Returns:
        list: A list of strings representing the requested verses of the song.

    This function generates the lyrics of 'The Twelve Days of Christmas' song based on the
    specified range of verses. It starts from the 'start_verse' and continues up to the 'end_verse'.
    The verses include the cumulative gifts received on each day.

    Example:
        start_verse = 1
        end_verse = 3
        recite(start_verse, end_verse) returns:
        [
            "On the first day of Christmas my true love gave to me: a Partridge in a Pear Tree.",
            "On the second day of Christmas my true love gave to me: two Turtle Doves, and a Partridge in a Pear Tree.",
            "On the third day of Christmas my true love gave to me: three French Hens, two Turtle Doves, and a Partridge in a Pear Tree."
        ]
    """
    # Initialize an empty list to store the song verses.
    result = []

    # Lists to store the ordinal numbers and corresponding gifts for each day.
    numbers = ["first", "second", "third", "fourth", "fifth", "sixth", "seventh", "eighth", "ninth", "tenth", "eleventh", "twelfth"]
    gifts = ["a Partridge in a Pear Tree.", "two Turtle Doves, ", "three French Hens, ", "four Calling Birds, ", "five Gold Rings, ", "six Geese-a-Laying, ", "seven Swans-a-Swimming, ", "eight Maids-a-Milking, ", "nine Ladies Dancing, ", "ten Lords-a-Leaping, ", "eleven Pipers Piping, ", "twelve Drummers Drumming, "]

    # Initialize the gift string with the first gift.
    str_gift = gifts[0]

    # Loop through the specified range of verses.
    for i in range(end_verse):
        # Construct the verse's ordinal number text.
        str_num = f"On the {numbers[i]} day of Christmas my true love gave to me: "

        # Concatenate the gift for the current day with the previous gifts.
        if i > 1:
            str_gift = gifts[i] + str_gift

        # Add "and" if it's the second day.
        if i == 1:
            str_gift = gifts[i] + "and " + str_gift

        # Append the complete verse to the result list.
        result.append(str_num + str_gift)

    # Return the requested verses, starting from start_verse.
    return result[start_verse - 1:]

File: javascript/twelve-days/twelve-days.js
export const recite = () => {
  const gifts = [
    "a partridge in a pear tree",
    "two turtle doves",
    "three French hens",
    "four calling birds",
    "five golden rings",
    "six geese a-laying",
    "seven swans a-swimming",
    "eight maids a-milking",
    "nine ladies dancing",
    "ten lords a-leaping",
    "eleven pipers piping",
    "twelve drummers drumming",
  ];

  const days = [
    "first",
    "second",
    "third",
    "fourth",
    "fifth",
    "sixth",
    "seventh",
    "eighth",
    "ninth",
    "tenth",
    "eleventh",
    "twelfth",
  ];

  let lyrics = "";
  for (let i = 0; i < 12; i++) {
    lyrics += `On the ${days[i]} day of Christmas, my true love gave to me\n`;
    for (let j = i; j >= 0; j--) {
      if (j === 0 && i !== 0) {
        lyrics += `And ${gifts[j]}\n`;
      } else {
        lyrics += `${gifts[j]}\n`;
      }
    }
    if (i !== 11) {
      lyrics += "\n";
    }
  }

  return lyrics;
};
